### PR TITLE
Adding app template test cases

### PR DESCRIPTION
**Pull request by coryodaniel**
When testing, we need to know _what_ failed in application templates, not if app templates elsewhere are rendering correctly or not.

* [added] two happy path test cases
  * `renders-successfully` is a basic happy path test of interpolation
  * `renders-dependencies` is a test on .Dependencies being interpolated
* [removed] copy/pasta from application-templates
* [added] liquid whitespace trimming (we'll need to do these in our templates)
* [added] diff output on difference (doesnt look great, best output was HTML, not positive how to get the diffs to show correctly when logged from `go test`

### CODE REVIEW

File: pkg/template/render_test.go [hunk 1]
```diff
@@ -70,9 +70,60 @@ func TestAppFromTemplate(t *testing.T) {
...
 			if gotMD5 != wantMD5 {
-				// TODO: need to print out which file is different
 				t.Errorf("got %v, want %v", gotMD5, wantMD5)
+				walkAndCompare(tc.wantPath, templateData.OutputDir)
+			}
+		})
+	}
+}
+
+func walkAndCompare(wantDir string, gotDir string) {
```
Comment (xpositivityx) [inline pkg/template/render_test.go:80]: SIIIIIck

File: pkg/template/testdata/application-templates/renders-dependencies/massdriver.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    hostname:
+      type: string
+    app_env:
+      type: string
+
+dependencies:
+  database:
+    type: massdriver/postgresql-authentication
+    required: true
+<md- range $key, $art:= .Dependencies md>
```
Comment (WillBeebe) [inline pkg/template/testdata/application-templates/renders-dependencies/massdriver.yaml:19]: great test case, all we need

**Review by xpositivityx — APPROVED** [review]

File: pkg/template/render_test.go (hunk 1 above)
Comment (WillBeebe) [inline pkg/template/render_test.go:80]: Thank you, this is much more helpful that the directory compare.

**Comment by xpositivityx** [comment]
This is great. Good TDD means light tests with limited code bloat.

**Review by WillBeebe — APPROVED** [review]
LGTM, thanks for paying down debt

File: pkg/template/render_test.go (hunk 1 above)
Comment (coryodaniel) [reply pkg/template/render_test.go:80]: The diff tool is pretty powerful, there’s just something about how go test handles std out that it loses the fancy formatting.

**Pull request merged**